Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM python:3.7-slim-buster as base
+RUN apt-get update \
+&& apt-get install -y curl \
+&& curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV PATH="${PATH}:/root/.poetry/bin"
+COPY ./todo-app ./todo-app 
+EXPOSE 5000
+
+FROM base as production
+ENTRYPOINT cd ./todo-app/ && poetry install && poetry run gunicorn --bind 0.0.0.0:5000 wsgi:app
```

Comment:
- Rather than `cd ./todo-app` you should have a separate Dockerfile command to set [WORKDIR](https://docs.docker.com/engine/reference/builder/#workdir)
E.g.
```
WORKDIR /todo-app
COPY ./todo-app ./

FROM base as production
ENTRYPOINT poetry install && poetry run gunicorn --bind 0.0.0.0:5000 wsgi:app
```

- It would be better to `RUN poetry install` rather than include it in the entrypoint, so it's done at build time rather than runtime. But arguably you want it in the entrypoint for the dev image so that you can modify dependencies and re-run it with the bind mount without a rebuild step. But you're adding a delay to every other run

- To improve the speed of rebuilds, you could copy just the poetry files, run poetry install and then copy the rest. So if you modify a python file for example, Docker won't need to re-run poetry install.

Reply:
Hey Hugh yes for some reason it was failing silently when i tried to do the install in the dockerfile, using WORKDIR has worked a treat though. I had always thought that var was more about what dir you entered to when running the image. Thanks :)